deps/rabbit/src/rabbit_db_queue.erl

```diff
@@ -1129,6 +1113,24 @@ do_delete_transient_queues_in_khepri(Qs, FilterFun) ->
             Error
     end.
 
+do_delete_transient_queues_in_khepri_tx([], Acc) ->
+    {ok, Acc};
+do_delete_transient_queues_in_khepri_tx([{Path, Vsn, QName} | Rest], Acc) ->
+     %% Also see `delete_in_khepri/2'.
+     VersionedPath = khepri_path:combine_with_conditions(
+                       Path, [#if_payload_version{version = Vsn}]),
```

Comment:
That’s a good idea to use the payload version here, given the read is outside of the transaction!

But now, I wonder if we need the transaction at all :-)

Reply:
I think the transaction would only be necessary for the sake of performance - it's faster to send the one transaction command than many small delete commands (since Ra handles them in serial). We might be able to make it work with async commands though, or a batch API.

The hope with some of the changes in 0.17 is to remove the transaction for regular queue deletion too (rather than just the transient ones here) since that will save resources + be faster

Reply:
I see, good point.